From c0bff71057ed103ec509977f7dcfbe124924aa4a Mon Sep 17 00:00:00 2001
From: tstegema <tstegema@90b65887-3827-0410-9a23-83215b262276>
Date: Thu, 12 Nov 2020 19:32:11 +0000
Subject: [PATCH 1034/1077] MANTIS 0059171 changed: DeviceHelp UTF-8 CORBA

use system property corba.encoding for open orb native code set
use system property file.encoding as fallback

add eclipse project settings for open orb



git-svn-id: svn://localhost/SelfServiceCommon/trunk@1606 90b65887-3827-0410-9a23-83215b262276
---
 .../Massai/java/OpenORB-1.3.1/.classpath      | 13 +++++++++++
 .../Massai/java/OpenORB-1.3.1/.project        | 23 +++++++++++++++++++
 .../iiop/CodeSetDatabaseInitializer.java      |  6 ++++-
 3 files changed, 41 insertions(+), 1 deletion(-)
 create mode 100644 SelfServiceCommon/Massai/java/OpenORB-1.3.1/.classpath
 create mode 100644 SelfServiceCommon/Massai/java/OpenORB-1.3.1/.project

diff --git a/SelfServiceCommon/Massai/java/OpenORB-1.3.1/.classpath b/SelfServiceCommon/Massai/java/OpenORB-1.3.1/.classpath
new file mode 100644
index 00000000..df7ec41d
--- /dev/null
+++ b/SelfServiceCommon/Massai/java/OpenORB-1.3.1/.classpath
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<classpath>
+	<classpathentry kind="src" path="OpenORB/src/main"/>
+	<classpathentry kind="src" path="OpenORB/build/main"/>
+	<classpathentry kind="src" path="OpenORB/build/gensrc/main"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.junit.JUNIT_CONTAINER/4"/>
+	<classpathentry kind="lib" path="tools/lib/avalon-framework.jar"/>
+	<classpathentry kind="lib" path="tools/lib/logkit.jar"/>
+	<classpathentry kind="lib" path="tools/lib/excalibur-configuration.jar"/>
+	<classpathentry kind="lib" path="tools/lib/xerces.jar"/>
+	<classpathentry kind="output" path="target/classes"/>
+</classpath>
diff --git a/SelfServiceCommon/Massai/java/OpenORB-1.3.1/.project b/SelfServiceCommon/Massai/java/OpenORB-1.3.1/.project
new file mode 100644
index 00000000..d949b76b
--- /dev/null
+++ b/SelfServiceCommon/Massai/java/OpenORB-1.3.1/.project
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<projectDescription>
+	<name>OpenORB</name>
+	<comment></comment>
+	<projects>
+	</projects>
+	<buildSpec>
+		<buildCommand>
+			<name>org.eclipse.jdt.core.javabuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+		<buildCommand>
+			<name>net.sf.eclipsecs.core.CheckstyleBuilder</name>
+			<arguments>
+			</arguments>
+		</buildCommand>
+	</buildSpec>
+	<natures>
+		<nature>org.eclipse.jdt.core.javanature</nature>
+		<nature>net.sf.eclipsecs.core.CheckstyleNature</nature>
+	</natures>
+</projectDescription>
diff --git a/SelfServiceCommon/Massai/java/OpenORB-1.3.1/OpenORB/src/main/org/openorb/iiop/CodeSetDatabaseInitializer.java b/SelfServiceCommon/Massai/java/OpenORB-1.3.1/OpenORB/src/main/org/openorb/iiop/CodeSetDatabaseInitializer.java
index a029691f..f97fa2b7 100644
--- a/SelfServiceCommon/Massai/java/OpenORB-1.3.1/OpenORB/src/main/org/openorb/iiop/CodeSetDatabaseInitializer.java
+++ b/SelfServiceCommon/Massai/java/OpenORB-1.3.1/OpenORB/src/main/org/openorb/iiop/CodeSetDatabaseInitializer.java
@@ -225,7 +225,11 @@ public class CodeSetDatabaseInitializer
     private CodeSetComponentInfo initCodeSets( boolean server )
     {
         // get the system encoding from the system properties (should be canonicalized)
-        String sysenc = System.getProperty( "file.encoding" );
+        String sysenc = System.getProperty( "corba.encoding" );
+        if(sysenc == null) 
+        {
+          sysenc = System.getProperty( "file.encoding" );
+        }
 
         // get the corresponding codeset id for the system encoding
         ArrayList al;
-- 
2.41.0.windows.1

